File: c++/Neetcode_150/1_Arrays&Hashing/1_contains_duplicate.cpp
//Return true if any value in array appears atleast twice ; else return false.

//S1

bool containsDuplicate(vector<int>& nums) {
       sort(nums.begin(), nums.end());
       for(int i=0; i<nums.size()-1; i++){
           if(nums[i] == nums[i+1]) return true;
       }
       return false;
    }

//S2

bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> st;
        for(auto n: nums){
            if(st.find(n) != st.end()) return true;
            else st.insert(n);
        }
        return false;
    }

File: js/Neetcode_150/1_Arrays&Hashing/1_contains_duplicate.js
//Return true if any value in array appears atleast twice ; else return false.

//S1:
var containsDuplicate = function(nums) {
    let set = new Set()
    for(let n of nums){
        if(set.has(n)) return true
        else set.add(n)
    }
    return false
};

//S2:
var containsDuplicate = function(nums) {
    nums.sort()
    for(let i=1; i<nums.length; i++){
        if(nums[i] == nums[i-1]) return true
    }
    return false
};